mars/dataframe/datastore/to_vineyard.py

```diff
@@ -0,0 +1,186 @@
+    import vineyard
+    from vineyard.data.dataframe import make_global_dataframe
+    from vineyard.data.utils import to_json
+except ImportError:
+    vineyard = None
+
+
+class DataFrameToVineyardChunk(DataFrameOperand, DataFrameOperandMixin):
+    _op_type_ = OperandDef.DATAFRAME_STORE_VINEYARD_CHUNK
+
+    # vineyard ipc socket
+    _vineyard_socket = StringField('vineyard_socket')
+
+    # vineyard object id
+    _vineyard_object_id = StringField('vineyard_object_id')
+
+    def __init__(self, vineyard_socket=None, dtypes=None, **kw):
+        super().__init__(_vineyard_socket=vineyard_socket, _dtypes=dtypes,
+                         _output_types=[OutputType.dataframe], **kw)
+
+    def __call__(self, df):
+        return self.new_dataframe([df], shape=(0, 0), dtypes=df.dtypes,
+                                  index_value=df.index_value, columns_value=df.columns_value)
+
+    @property
+    def vineyard_socket(self):
+        return self._vineyard_socket
+
+    @property
+    def vineyard_object_id(self):
+        return self._vineyard_object_id
+
+    @classmethod
+    def _get_out_chunk(cls, op, in_chunk):
+        chunk_op = op.copy().reset_key()
+        out_chunk_shape = (np.nan,) * in_chunk.ndim
+        return chunk_op.new_chunk([in_chunk], shape=out_chunk_shape,
+                                  index=in_chunk.index)
+
+    @classmethod
+    def _process_out_chunks(cls, op, out_chunks):
+        merge_op = DataFrameToVinyardStoreMeta(
+                vineyard_socket=op.vineyard_socket,
+                chunk_shape=op.inputs[0].chunk_shape,
+                shape=op.inputs[0].shape,
+                dtypes=op.inputs[0].dtypes)
+        return merge_op.new_chunks(out_chunks, shape=(1,),
+                                   index=(0,) * out_chunks[0].ndim)
+
+    @classmethod
+    def tile(cls, op):
+        out_chunks = []
+        scheduling_hint = SchedulingHint(not_fuseable=True)
+        for chunk in op.inputs[0].chunks:
+            chunk_op = op.copy().reset_key()
+            chunk_op.scheduling_hint = scheduling_hint
+            out_chunk = chunk_op.new_chunk([chunk], shape=chunk.shape, dtypes=chunk.dtypes,
+                                           index_value=chunk.index_value,
+                                           columns_value=chunk.columns_value,
+                                           index=chunk.index)
+            out_chunks.append(out_chunk)
+        out_chunks = cls._process_out_chunks(op, out_chunks)
+
+        in_df = op.inputs[0]
+        new_op = op.copy().reset_key()
+        return new_op.new_dataframes(op.inputs, shape=op.inputs[0].shape,
+                                     dtypes=in_df.dtypes,
+                                     index_value=in_df.index_value,
+                                     columns_value=in_df.columns_value,
+                                     chunks=out_chunks, nsplits=((np.prod(op.inputs[0].chunk_shape),),))
+
+    @classmethod
+    def execute(cls, ctx, op):
```

Comment:
Quite a few lines in this function are not covered, please check why.

Reply:
I think that is because we doesn't run the test in `test_local.py` with with `--forked` thus we cannot capture the execution of the `execute` method.

I will try to add an extra run to ensure it been captured.